### PR TITLE
Keep vertical spaces between items or statements within range 

### PR DESCRIPTION
**Pull request by topecongiro**
cc https://github.com/rust-lang-nursery/fmt-rfcs/issues/57.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -651,6 +651,10 @@ create_config! {
         "Add trailing semicolon after break, continue and return";
     match_block_trailing_comma: bool, false, false,
         "Put a trailing comma after a block based match arm (non-block arms are not affected)";
+    blank_lines_upper_bound: usize, 1, false,
```
Comment (nrc) [inline src/config.rs]: The default here should be 2.

**Comment by topecongiro** [comment]
Thank you for taking time to review this!

I read [your comment](https://github.com/rust-lang-nursery/fmt-rfcs/issues/57#issuecomment-291303071) and it states that we should have either 0 or 1 blank line between items. Do you mean that the config values should represent the exclusive bound?

**Comment by topecongiro** [comment]
This PR does not handle blank lines around comments. The blank lines in the example below will be kept untouched:
```rust
fn foo()



// comment



fn bar() {}
```
I am going to update this PR, add more tests.

**Comment by nrc** [comment]
> I read your comment and it states that we should have either 0 or 1 blank line between items. Do you mean that the config values should represent the exclusive bound?

You are correct, sorry - my memory was not as good as I thought :-)

**Comment by topecongiro** [comment]
Updated with handling of comments and tests. My apologies for messy commit history.

**Comment by nrc** [comment]
Thank you!

**Pull request merged**